examples/multiplexer/host/src/main.rs

```diff
@@ -85,19 +97,34 @@ async fn main() -> eyre::Result<()> {
 
     // Generate the proof for the given program and input.
     let (pk, vk) = client.setup(ELF);
-    let mut proof = client.prove(&pk, stdin).run().unwrap();
+    println!("vkey: {:?}", vk.bytes32());
+    let proof = client.prove(&pk, stdin).plonk().run().unwrap();
     println!("generated proof");
 
-    // Read the state root, and verify it.
-    let client_state_root = proof.public_values.read::<B256>();
-    assert_eq!(client_state_root, state_root);
+    // proof.save("proof-with-pis.bin").expect("saving proof failed");
 
-    // Read the output, in the form of a bunch of exchange rates.
-    //
-    // Note that this output is read from values commited to in the program using
-    // `sp1_zkvm::io::commit`.
-    let result = proof.public_values.read::<Vec<U256>>();
-    println!("Got these rates: \n{:?}%", result);
+    // Save the proof to plonk-proof.bin
+    let mut proof_file = File::create("plonk-proof.bin")?;
+    proof_file.write_all(&proof.bytes())?;
+
+    // Save the public values to public-values.bin
+    let mut public_values_file = File::create("public-values.bin")?;
+    public_values_file.write_all(proof.public_values.as_slice())?;
+
+    // Read the public values, and deserialize them.
+    let public_vals = MultiplexerOutput::abi_decode(proof.public_values.as_slice(), true)?;
+    let contract_public_values = public_vals.contractPublicValues;
+
+    // Read the block hash, and verify that it's the same as the one inputted.
+    assert_eq!(contract_public_values.blockHash, block_hash);
+
+    // Print the fetched rates.
+    let rates = getRatesCall::abi_decode_returns(&contract_public_values.contractOutput, true)?._0;
+    println!("Got these rates: \n{:?}", rates);
+
+    // Print the timestamp and block number.
+    println!("Block timestamp: {}", public_vals.blockTimestamp);
+    println!("Block number: {}", public_vals.blockNumber);
```

Comment:
Remove the user specific logic here so this example is clean and straightforward